src/com/projectkorra/projectkorra/region/RegionProtection.java

```diff
@@ -84,17 +84,13 @@ public static Map<JavaPlugin, RegionProtectionHook> getActiveProtections() {
      * @return True if the region is protected by other plugins
      */
     public static boolean isRegionProtected(@NotNull Player player, @Nullable Location location, @Nullable CoreAbility ability) {
-        if (!BLOCK_CACHE.containsKey(player.getName())) {
-            BLOCK_CACHE.put(player.getName(), new ConcurrentHashMap<>());
-        }
+        final String playerName = player.getName();
+        final Block block = location != null ? location.getBlock() : player.getLocation().getBlock();
+        final Map<Block, BlockCacheElement> blockMap = BLOCK_CACHE.computeIfAbsent(playerName, name -> new ConcurrentHashMap<>());
```

Comment:
Now that BLOCK_CACHE elements can be removed we need to turn checking the existence of a key and getting the value of the item from the Map into a single atomic action to avoid a Null Pointer Exception in the event that the CleanUp happens at the same time as an isRegionProtected call. This is also the major downside of this change. It introduces an opportunity for blocks to not get properly cached. If a block is cached at the same time that a cleanup happens, it is possible that the cached blocks are getting added to a Map that has already been removed from BLOCK_CACHE